packages/frameworks-sveltekit/src/lib/index.ts

```diff
@@ -277,6 +363,9 @@ export function SvelteKitAuth(
 
     event.locals.auth ??= () => auth(event, _config)
     event.locals.getSession ??= event.locals.auth
+    event.locals.signIn ??= (provider, options, authorizationParams) =>
+      signIn(provider, options, authorizationParams, _config, event)
+    event.locals.signOut ??= async (options) => signOut(options, _config, event)
```

Comment:
I feel like this should not be part of the `event`. :thinking: How/where would you call these methods?

A more appropriate solution would be to return more than `Handle` here:

https://github.com/nextauthjs/next-auth/blob/70efa75ec2b1b77fa2920d7f4fe031f108e2b9b3/packages/frameworks-sveltekit/src/lib/index.ts#L357

and introduce the concept of `auth.ts` similar to `next-auth`.

Reply:
If returning more than `Handle` it would force users to make their `src/hooks.server.ts` more complex though. Also if exporting the `signIn`/`signOut` methods from there folks woudl have to `import { signIn } from './src/hooks.server.ts'` wherever they'd want to use it. Feels a bit weird importing from the hooks file. That'd be like importing from the `middleware.ts` to anywhere in your Next.js 13+ app afaik :thinking:

Reply:
We would introduce the concept of `auth.ts`, so `hooks.server.ts` would become `export { handle } from "./auth"` similar to `export { auth as middleware } from "auth" in NextAuth.js

Reply:
Ahh right okay that has much better ergonomics

Reply:
I think we can use it like this, no? 
```js
import type { LayoutServerLoad } from "./$types"

export const load: LayoutServerLoad = async (event) => {
  await event.locals.signIn("credentials", { password: "pw" }) // should handle redirect
}

```

Reply:
So I added some usage examples in the dev app, let me know what you think. In SvelteKit, we can assign to the `locals`property and just use in many context: server only with `+server.ts`, layout load or page load.